syntax = "proto3";

package datacloud.console.v1;

import "datacloud/v1/paging.proto";
import "datacloud/console/v1/cluster.proto";

option go_package = "a.yandex-team.ru/cloud/mdb/datacloud/private_api/datacloud/console/v1;console";
option java_package = "doublecloud.priv.console.v1";
option java_outer_classname = "DCCS";

service ClusterService {
    // Retrieves a list of clusters that belong to the specified project.
    rpc List (ListClustersRequest) returns (ListClustersResponse);
    rpc ProjectByClusterID (ProjectByClusterIDRequest) returns (ProjectByClusterIDResponse);
}

message ListClustersRequest {
    // ID of the project to list clusters in.
    string project_id = 1;

    // Paging information of the request
    datacloud.v1.Paging paging = 2;
}

message ListClustersResponse {
    // List of cluster resources.
    repeated Cluster clusters = 1;

    // Pagination information of the response
    datacloud.v1.NextPage next_page = 2;
}

message ProjectByClusterIDRequest {
    // ID of the cluster to get project_id for.
    string cluster_id = 1;
}

message ProjectByClusterIDResponse {
    // Corresponding project_id for requested cluster.
    string project_id = 1;
}
